connectors/sources/postgresql.py

```diff
@@ -83,6 +83,7 @@ class PostgreSQLAdvancedRulesValidator(AdvancedRulesValidator):
         "properties": {
             "tables": {"type": "array", "minItems": 1},
             "query": {"type": "string", "minLength": 1},
+            "id_columns": {"type": "array", "minItems": 0},
```

Comment:
Why `minItems: 0`? So it's not required, right?

Reply:
No, we need to keep `minItems: 1` for `id_columns` as its optional field, but if the user added that, we need at least one column to generate IDs. I have updated the code. Thanks.